modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -296,13 +301,78 @@ object JacksonGenerator {
         case _ if parents.length == 1                   => Target.pure(parents.headOption)
         case _                                          => Target.pure(None)
       }
+
+      discriminators                             = parents.flatMap(_.discriminators)
+      discriminatorNames                         = discriminators.map(_.propertyName).toSet
+      parentParams                               = parentOpt.toList.flatMap(_.params)
+      parentParamNames                           = parentParams.map(_.name)
+      (parentRequiredTerms, parentOptionalTerms) = sortParams(parentParams)
+      parentTerms                                = parentRequiredTerms ++ parentOptionalTerms
+
+      discriminatorValues <- parentTerms
+        .flatMap({ term =>
+          discriminators.find(_.propertyName == term.propertyName).map((term, _))
+        })
+        .traverse({
+          case (term, discriminator) =>
+            val discriminatorValue = discriminator.mapping
+              .collectFirst({ case (value, elem) if elem.name == clsName => value })
+              .getOrElse(clsName)
+
+            def parseLiteral(parser: String => Expression, friendlyName: String): Target[Expression] =
+              Try(parser(discriminatorValue)).fold(
+                t => Target.raiseError[Expression](s"Unable to parse '$discriminatorValue' as '$friendlyName': ${t.getMessage}"),
+                Target.pure[Expression]
+              )
+
+            val discriminatorValueExpr = term.rawType.tpe match {
+              case Some(tpe @ "string") =>
+                term.rawType.format match {
+                  case Some("date") | Some("date-time") | Some("byte") | Some("binary") =>
+                    Target.raiseError[Expression](
+                      s"Unsupported discriminator type '$tpe' with format '${term.rawType.format.getOrElse("unknown")}' for property '${term.propertyName}'"
+                    )
+                  case _ => Target.pure[Expression](new StringLiteralExpr(discriminatorValue))
+                }
+              case Some(tpe @ "boolean") => parseLiteral(x => new BooleanLiteralExpr(x.toBoolean), tpe)
+              case Some(tpe @ "integer") =>
+                term.rawType.format match {
+                  case Some(fmt @ "int32") => parseLiteral(x => new IntegerLiteralExpr(x.toInt), fmt)
+                  case Some(fmt @ "int64") => parseLiteral(x => new LongLiteralExpr(x.toLong), fmt)
+                  case Some(fmt) =>
+                    Target.raiseError[Expression](s"Unsupported discriminator type '$tpe' with format '$fmt' for property '${term.propertyName}'")
+                  case None =>
+                    parseLiteral({ x =>
+                      new BigInteger(x); new ObjectCreationExpr(null, BIG_INTEGER_FQ_TYPE, new NodeList(new StringLiteralExpr(discriminatorValue)))
```

Comment:
oh jeez. Could we not serialize the `BigInteger` back into something that can be stuffed into `ObjectCreationExpr`, lest some linter start calling it out as a non-unit expression?

Reply:
I guess could do `new ObjectCreationExpr(..., new BigInteger(x).toString())`?  in theory it shouldn't lose precision, since that's the point of `BigInteger`, yeah?

Reply:
Did this ^. Seems to be ok, checked some numbers with very very very very very large numbers both to the right and left of the decimal, as well as small, and there's no loss in precision.